grails-app/controllers/org/pih/warehouse/JsonController.groovy

```diff
@@ -1822,6 +1816,23 @@ class JsonController {
             throw new IllegalArgumentException("Start and end date are required")
         }
     }
+
+    def checkIfProductFieldRemoved = {
```

Comment:
The guts of this should probably go in a method on the ProductType class but we can leave that for now.